### PR TITLE
Upgrade to Apache Beam, version 2.0.0

### PR DESCRIPTION
**Pull request by jasonkuster**
R: @davorbonaci @dhalperi

### CODE REVIEW

File: maven-archetypes/examples-java8/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -68,27 +69,33 @@
...
-        <version>2.4.1</version>
+        <version>3.0.0</version>
         <executions>
           <execution>
             <phase>package</phase>
             <goals>
               <goal>shade</goal>
             </goals>
             <configuration>
-              <finalName>${project.artifactId}-bundled-${project.version}</finalName>
```
Comment (davorbonaci) [inline maven-archetypes/examples-java8/src/main/resources/archetype-resources/pom.xml]: revert?
pr to beam?

File: maven-archetypes/examples/src/main/resources/archetype-resources/pom.xml [hunk 2]
```diff
@@ -68,27 +69,33 @@
...
-        <version>2.4.1</version>
+        <version>3.0.0</version>
         <executions>
           <execution>
             <phase>package</phase>
             <goals>
               <goal>shade</goal>
             </goals>
             <configuration>
-              <finalName>${project.artifactId}-bundled-${project.version}</finalName>
```
Comment (davorbonaci) [inline maven-archetypes/examples/src/main/resources/archetype-resources/pom.xml]: revert?
push a pr to beam?

File: pom.xml [hunk 3]
```diff
@@ -103,7 +103,7 @@
     <timestamp>${maven.build.timestamp}</timestamp>
     <maven.build.timestamp.format>yyyy-MM-dd HH:mm</maven.build.timestamp.format>
 
-    <beam.version>0.6.0</beam.version>
+    <beam.version>2.0.0</beam.version>
 
     <dataflow.release_name>Google Cloud Dataflow SDK for Java</dataflow.release_name>
     <dataflow.container_version>${project.version}-20170317</dataflow.container_version>
```
Comment (dhalperi) [inline pom.xml]: For the code at HEAD with this PR to work, we would need a new (dated) container.

Options:
* build a new 2.0.0-beta4-SNAPSHOT-20170522 container for this PR only.
* deal with breakage on the assumption that maven release:prepare is going to immediately change the version to (2.0.0-SNAPSHOT) or (2.1.0-SNAPSHOT) and `HEAD` will be broken for a short time.

File: pom.xml [hunk 4]
```diff
@@ -103,7 +103,7 @@
     <timestamp>${maven.build.timestamp}</timestamp>
```
Comment (dhalperi) [inline pom.xml]: is Maven 3.0.0 above still accurate? (maybe? probably?)

File: pom.xml [hunk 5]
```diff
@@ -103,7 +103,7 @@
...
     <maven.build.timestamp.format>yyyy-MM-dd HH:mm</maven.build.timestamp.format>
 
-    <beam.version>0.6.0</beam.version>
+    <beam.version>2.0.0</beam.version>
```
Comment (dhalperi) [inline pom.xml]: I believe the dependency version numbers in this pom.xml need to be checked against the ones in the Beam root pom.xml. As a spot-check, I saw surefire here is at 2.19.1.

File: maven-archetypes/examples-java8/src/main/resources/archetype-resources/pom.xml (hunk 1 above)
Comment (jasonkuster) [inline maven-archetypes/examples-java8/src/main/resources/archetype-resources/pom.xml]: Reverted. PR to beam to follow.

File: maven-archetypes/examples/src/main/resources/archetype-resources/pom.xml (hunk 2 above)
Comment (jasonkuster) [inline maven-archetypes/examples/src/main/resources/archetype-resources/pom.xml]: Reverted. PR to beam to follow.

File: pom.xml (hunk 3 above)
Comment (jasonkuster) [inline pom.xml]: We decided that option 2 was what we were going to do. I will add this to the documentation.

File: pom.xml (hunk 4 above)
Comment (jasonkuster) [inline pom.xml]: Updated to maven 3.2 per beam pom.xml.

File: pom.xml (hunk 5 above)
Comment (jasonkuster) [inline pom.xml]: Done.

**Pull request merged**